scopesim/utils.py

```diff
@@ -555,27 +555,73 @@ def from_rc_config(item):
     return from_currsys(item, rc.__config__)
 
 
-def check_keys(input_dict, required_keys, action="error", all_any="all"):
-    """Check to see if all/any of the required keys are present in a dict."""
-    if isinstance(input_dict, (list, tuple)):
-        input_dict = {key: None for key in input_dict}
+def check_keys(input_dict: Union[Mapping, Iterable],
+               required_keys: Set,
+               action: str = "error",
+               all_any: str = "all") -> bool:
+    """
+    Check to see if all/any of the required keys are present in a dict.
+
+    .. versionchanged:: v0.8.0
+        The `required_keys` parameter should now be a set.
+
+    Parameters
+    ----------
+    input_dict : Union[Mapping, Iterable]
+        The mapping to be checked.
+    required_keys : Set
+        Set containing the keys to look for.
+    action : {"error", "warn", "warning"}, optional
+        What to do in case the check does not pass. The default is "error".
+    all_any : {"all", "any"}, optional
+        Whether to check if "all" or "any" of the `required_keys` are present.
+        The default is "all".
+
+    Raises
+    ------
+    ValueError
+        Raised when an invalid parameter was passed or when `action` was set to
+        "error" (the default) and the `required_keys` were not found.
+
+    Returns
+    -------
+    keys_present : bool
+        ``True`` if check succeded, ``False`` otherwise.
+
+    """
+    # Checking for Set from collections.abc instead of builtin set to allow
+    # for any duck typing (e.g. dict keys view or whatever)
+    if not isinstance(required_keys, Set):
+        logger.warning("required_keys should implement the Set protocol, "
+                       "found %s instead.", type(required_keys))
+        required_keys = set(required_keys)
+
+    # First assume it's a Mapping, otherwise attempt to convert to set.
+    try:
+        input_keys = input_dict.keys()
+    except AttributeError:
+        try:
+            input_keys = set(input_dict)
+        except Exception as err:
+            raise ValueError("input_dict must be mapping or iterable, but was "
+                             f"{type(input_dict)}") from err
```

Comment:
It seems that this conversion to set is not necessary, as `issubset()`  and `isdisjoint()` seem to work as expected with a dictionary or list as argument. So I think it would be cleaner to simply remove this code
```suggestion
```

Reply:
Hm, I clearly remember there was a reason I put this in, i.e. I ran into a case where it didn't work. I'll see if I can find that again...

Reply:
Ok, it seems to be fine to just rely on the internal set conversion done by these set methods. I think I originally used operators instead of the methods (because there's more than one obvious way to do this kind of thing...), which would result in TypeErrors.